### PR TITLE
Create IDX Skeleton Page

### LINKED ISSUE #6422: Create IDX Skeleton Pages for both Canonical and AMP
**Is your feature request related to a problem? Please describe.**
We want to render Index (IDX) pages in Simorgh which should replicate the current Front Page behaviour. 
Current examples in PAL: 
- https://www.bbc.com/ukrainian/ukraine_in_russian
- https://www.bbc.com/persian/afghanistan
This ticket is about starting off this work by just making a page container and making the routes.

**Describe the solution you'd like**
We should begin with the creation of a skeleton page with the routes set up at /persian/afghanistan, /persian/afghanistan.amp, /ukrainian/ukraine_in_russian and /ukrainian/ukraine_in_russian.amp and for Canonical and AMP respectively. 
- This will require an IDX page container which can just return a div for the time being.
- This will also require adding the routes to `mozart-routing`, covered by https://github.com/bbc/simorgh-infrastructure/issues/992

**Post-refinement comments**
1) Regarding routing:  Are there going to be more IDX assets in the future? @tochwill confirmed with @jimjohnsonrollings that no future IDX assets will be made so we only need to route these two pages.

2) The goal is just to get a page on test, the contents, layout, etc. does not need to be considered at the moment.

3) **Technical steps involved:**
 - Make a new route for both IDX pages in Simorgh
   - Create a route which doesn't fall under CPS assets due to having a unique regex for just these routes: 
**/persian/afghanistan
/persian/afghanistan.amp
/ukrainian/ukraine_in_russian
/ukrainian/ukraine_in_russian.amp**
   - Using mostRead as an example in routes, we need a similar export for IDX which specifies `pagetype` and returns a 200 response in the `getInitialData` function
   - ~~Check if IDX pages has a metadata object inside the response. We shouldn't need to do anything as long as Ares is giving us the data~~ @tochwill checked this and confirmed we do have metadata being returned

 - Make a page container using higher order components (eg. withPageWrapper, withContexts) alongside the other page types.
   - Create a snapshot test just to check that it is rendering an empty page. This can be updated later when content is added.

Mozart routing and STM changes will then need to be made in order to see this page on the test environment. This is covered by this issue: https://github.com/bbc/simorgh-infrastructure/issues/992

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by hotinglok**
Resolves #6422 

**Overall change:**
Created simorgh routes and page type/container for IDXs

**Code changes:**

- Added Regex for `persian/afghanistan` and `ukrainian/ukraine_in_russian`.
- Added routing for IDX pages.
- Created an empty `IDXPage` page type.
- Created a snapshot (empty for now) for IDX pages.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/routes/index.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import mostRead from './mostRead';
 import onDemandRadio from './onDemandRadio';
 import error from './error';
 import errorNoRouteMatch from './errorNoRouteMatch';
+import index from './idx';
```
Comment (hotinglok) [inline src/app/routes/index.js]: Was getting errors renaming the folder to `index` so I've left it as `idx`

Comment (EinsteinNjoroge) [reply src/app/routes/index.js]: 🤔 Would it be worth renaming `index` to `idx` just for clarity?
In most cases, the name`index` indicates that it's the default page/script

**Review by DenisHdz — COMMENTED** [review]
Overall looks good! Could you also add an `index.stories.jsx` file so we can have this page type in Storybook?

idx-ukranian

File: src/app/routes/index.js (hunk 1 above)
Comment (tochwill) [reply src/app/routes/index.js]: Clashes with 'index' is a fun one! `idx` may be better because of this!

File: src/app/pages/IndexPage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const IndexPage = () => {
+  return (
+    <main>
```
Comment (DenisHdz) [inline src/app/pages/IndexPage/index.jsx]:
```suggestion
    <main role="main">
```

**Comment by tochwill** [comment]
Could we as rule not use 'Resolves #6430 ' (i.e. the parent issues) in PRs? This will cause github to close the parent issue when we merge each individual PR  😁

File: src/app/pages/IndexPage/index.jsx (hunk 2 above)
Comment (DenisHdz) [reply src/app/pages/IndexPage/index.jsx]: Can we also add the `h1` as a part of the skeleton?

```suggestion
    <h1 id="content">Index Page</h1>
```

File: src/app/pages/IndexPage/index.test.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import IndexPage from '.';
+
+describe('IndexPage', () => {
+  shouldMatchSnapshot(
```
Comment (DenisHdz) [inline src/app/pages/IndexPage/index.test.jsx]: Worth having another `describe` to group the snapshots as we will be adding assertions later on.

```suggestion
describe('Snapshots', () => {
  shouldMatchSnapshot(
```

File: src/app/routes/index.js (hunk 1 above)
Comment (OlgaLyubin) [reply src/app/routes/index.js]: This has been done now:)

**Review by DenisHdz — APPROVED** [review]
Thanks for the updates 👍

File: src/app/routes/index.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ import mostRead from './mostRead';
 import onDemandRadio from './onDemandRadio';
 import error from './error';
 import errorNoRouteMatch from './errorNoRouteMatch';
+import IDX from './idx';
```
Comment (tochwill) [inline src/app/routes/index.js]:
```suggestion
import idx from './idx';
```

File: src/app/routes/index.js [hunk 5]
```diff
@@ -14,6 +15,8 @@ export default [
   liveRadio,
   mostRead,
   onDemandRadio,
+  IDX,
```
Comment (tochwill) [inline src/app/routes/index.js]:
```suggestion
  idx,
```

**Review by tochwill — COMMENTED** [review]
This looks really good, thanks! Sorry just an annoying tiny thing - could we update the casing of functions and files to match the rest of the project? `IdxPage` for React components, `idxPagePath` for regex etc...

File: src/app/routes/index.js [hunk 6]
```diff
@@ -14,6 +15,8 @@ export default [
...
   error,
-  errorNoRouteMatch,
+  errorNoRouteMatch, // When none of the above routes match, it will fall to this.
+  // Please ensure that when adding new pages, it is above this check in the array.
```
Comment (tochwill) [inline src/app/routes/index.js]:
```suggestion
  // Please ensure that when adding new pages, it is above this element in the array.
```

File: src/app/routes/idx/index.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { IdxPage } from '#pages';
+import { IdxPagePath } from '../utils/regex';
+
+export default {
+  path: IdxPagePath,
+  exact: true,
+  component: IdxPage,
+  getInitialData: () => Promise.resolve({ status: 200, pageData: {} }),
+  pageType: 'idx',
```
Comment (thekp) [inline src/app/routes/idx/index.js]: I think this should be in capitals to match what is in the RequestContext props

```suggestion
  pageType: 'IDX',
```

Comment (tochwill) [reply src/app/routes/idx/index.js]: I am now in a small rabbit hole looking at this, will report back when I've climbed out

Comment (tochwill) [reply src/app/routes/idx/index.js]: @thekp is correct though!

Comment (tochwill) [reply src/app/routes/idx/index.js]: What confused me isn't relevant to this conversation, I'll write it up somewhere. tl;dr I think we should have these defined as constants and use them in both places if they have to match, but we can do that in another PR.

Comment (tochwill) [reply src/app/routes/idx/index.js]: We already have a file with some constants in, but it's not being used for `pageType` in the routes files, or for the prop types in request context. I'll create an issue to change that :)

**Review by tochwill — APPROVED** [review]

**Pull request merged**